File: core/src/me/ferdz/evergreenacres/core/rendering/Textures.java
package me.ferdz.evergreenacres.core.rendering;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.TextureRegion;

public class Textures {
	
	public enum ItemTexture {
		WATERING_CAN(0, 0),
		HOE(0, 1);
		
		private int x, y;
		private ItemTexture(int x, int y) {
			this.x = x;
			this.y = y;
		}
		
		public int getX() {
			return x;
		}
		
		public int getY() {
			return y;
		}
	}
	private static TextureRegion[][] items;
	public static TextureRegion getTextureRegion(ItemTexture item) {
		if (items == null)
			items = TextureRegion.split(new Texture(Gdx.files.internal("homegrown/ui/items.png")), 16, 16);
		return items[item.x][item.y];
	}
	
	private static TextureRegion[][] uiTextures;
	public static TextureRegion[][] getUiTextures() {
		if (uiTextures == null) 
			uiTextures = TextureRegion.split(new Texture(Gdx.files.internal("homegrown/ui/ui.png")), 16, 16);

		return uiTextures;
	}
	
	public static TextureRegion getSelectedGroundTileTexture() {
		return getUiTextures()[3][0];
	}
}

File: core/src/me/ferdz/evergreenacres/rendering/Textures.kt
package me.ferdz.evergreenacres.rendering

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.TextureRegion

object Textures {
	enum class ItemTexture private constructor(x: Int, y: Int) {
		WATERING_CAN(0, 0),
		HOE(0, 1);

		val x: Int
		val y: Int

		init {
			this.x = x
			this.y = y
		}
	}
	
	@JvmStatic
	val items: Array<Array<TextureRegion>> by lazy {
		TextureRegion.split(Texture(Gdx.files.internal("homegrown/ui/items.png")), 16, 16)
	}
	
	@JvmStatic
	fun getTextureRegion(texture: ItemTexture): TextureRegion {
		return items[texture.x][texture.y];
	}
	
	@JvmStatic
	val uiTextures: Array<Array<TextureRegion>> by lazy {
		TextureRegion.split(Texture(Gdx.files.internal("homegrown/ui/ui.png")), 16, 16)
	}
	
	@JvmStatic
	val selectedGroundTileTexture: TextureRegion by lazy {
		uiTextures[3][0]		
	}
}